GMX-Bots/gmx5-detect-frontruns/src/agent.ts

```diff
@@ -0,0 +1,206 @@
+import {
+  BlockEvent,
+  Finding,
+  HandleBlock,
+  HandleTransaction,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  Log,
+  LogDescription,
+  ethers,
+  getEthersProvider,
+} from "forta-agent";
+import { NetworkManager } from "forta-agent-tools";
+interface NetworkData {
+  address: string;
+  num: number;
+}
+const data: Record<number, NetworkData> = {
+  42161: {
+    address: "0xabbc5f99639c9b6bcb58544ddf04efa6802f4064",
+    num: 1,
+  },
+  43114: {
+    address: "0x5f719c2f1095f7b9fc68a68e35b51194f4b6abe8",
+    num: 2,
+  },
+};
+let GMX_ROUTER_ADDRESS = { value: "" };
+const provider = getEthersProvider();
+export const initialize = (provider: ethers.providers.Provider) => async () => {
+  const networkManager = new NetworkManager(data, (await provider.getNetwork()).chainId);
+  GMX_ROUTER_ADDRESS.value = networkManager.get("address");
+};
+export const SWAP_EVENT =
+  "event Swap(address account, address tokenIn, address tokenOut, uint256 amountIn, uint256 amountOut)";
+
```

Comment:
I'd recommend moving the network data, ABIs and some constants ( `CALL_HISTORY_SIZE` for example) to a separate file.